### ISSUE TITLE
ResearchKit.h framework header imports gone wrong

### ISSUE THREAD

**Issue by malhal** [ResearchKit/ResearchKit#839]
This commit made a mistake in the umbrella header ResearchKit.h:

https://github.com/ResearchKit/ResearchKit/commit/f7d1591c19ad0c12a969f1dceeb8d41716079073

The imports in should be of the framework format:

```
#import <ResearchKit/ORKStep.h>
```

not

```
#import "ORKStep.h"
```

The commit states cocoa pods as the reason for the change, I'd be happy to advise on how to correct the cocoa pod configuration rather than break the header.

**Comment by rsanchezsaez** [comment]
Hi @malhal 

I think you're right, by looking at the rest of the Apple's Frameworks, they all use the *framework format*. I'm ok with changing them to this format.

However, the current direct format seems to work fine too: I built ResearchKit as a standalone framework, added the framework to a new project (not including ResearchKit source code), and it compiled and ran fine. Just to get to the bottom of this issue, in which way do you mean that using this header format is broken?

Thanks!
Ricardo

**Comment by malhal** [comment]
Hi @rsanchezsaez, yes the umbrella header ResearchKit.h should contain the framework format, as all public headers should contain the framework format, which it looks to me like most if not all already do. I'm not sure why you aren't encountering an error but right now if I include a pre-built version of the framework in a test project, or embed the ResearchKit project (with an appropriate include header dir build setting), either way when I do:

    #import <ResearchKit/ResearchKit.h>

I get the compiler error:
        `/Users/mh/Development/Test/ResearchTest/dependencies/ResearchKit/ResearchKit/ResearchKit.h:32:9: 'ORKTypes.h' file not found`

Which is accurate because it is importing "ORKTypes.h" and not <ResearchKit/ORKTypes.h>. 

How are you including Research Kit in your example project?

**Comment by rsanchezsaez** [comment]
Hey @malhal,

Sorry for my again late reply, I'm a bit busy. Still trying to get to the bottom of this.

For me it works with the (not completely correct) current headers. By manually adding a precompiled *ResearchKit* library to my project, leaving all the project settings by default, it compiles and runs:

<img width="1519" alt="screen shot 2016-10-29 at 9 01 19 pm" src="https://cloud.githubusercontent.com/assets/444313/19834411/ec1c5358-9e1a-11e6-98ae-6108a68fe423.png">

<img width="1519" alt="screen shot 2016-10-29 at 9 03 08 pm" src="https://cloud.githubusercontent.com/assets/444313/19834415/1dad8270-9e1b-11e6-972d-9a0086235dde.png">

I also tried using both variants:

    #import <ResearchKit/ResearchKit.h>

and

    @import ResearchKit;

Both worked.

---

Which version of Xcode are you using? Could you upload to GitHub a project of your making that is failing? I'd like to understand why it fails for you but not for me.

**Comment by malhal** [comment]
Hi @rsanchezsaez thanks for taking the time to reply and especially for the screenshots!

I did what you had, copied the framework into the project, that benefits from not having to add a framework search path because the project dir is already a default project dir. Since that worked then I realised my mistake was setting the "Header search paths" which I now know isn't needed for frameworks, that was breaking it in my case. Sorry for the trouble.
